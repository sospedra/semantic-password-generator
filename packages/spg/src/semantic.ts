import nlp from 'wink-nlp-utils'
import deburr from 'lodash.deburr'

export default function semantic(extract: string): string[] {
  return [
    nlp.string.removeHTMLTags,
    nlp.string.removeExtraSpaces,
    deburr,
    nlp.string.sentences,
  ]
    .reduce((memo, fn) => fn(memo), extract)
    .map(nlp.string.removePunctuations)
}
