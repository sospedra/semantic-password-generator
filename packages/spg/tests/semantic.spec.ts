import semantic from '../src/semantic'
import stubs from './stubs'

test('clean semantically common use cases', () => {
  stubs.forEach((sentence) => {
    expect(semantic(sentence)).toMatchSnapshot()
  })
})
