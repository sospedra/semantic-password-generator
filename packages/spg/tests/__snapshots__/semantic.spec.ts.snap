// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clean semantically common use cases 1`] = `
Array [
  "this is my long sentence to try and to check the length thiny",
]
`;

exports[`clean semantically common use cases 2`] = `
Array [
  "a bit more short but medium long",
]
`;

exports[`clean semantically common use cases 3`] = `
Array [
  "super short",
]
`;

exports[`clean semantically common use cases 4`] = `
Array [
  "itsn t a thing we ll understand",
]
`;

exports[`clean semantically common use cases 5`] = `
Array [
  "Bjork Gudmundsdottir OTF     b j ɜːr k   BYURK   Icelandic   pjoerk ˈkvʏdmʏntsˌtouʰtɪr    listen     born 21 November  1965  is an Icelandic singer  songwriter  record producer  actress  and DJ ",
  "Over her four decade career  she has developed an eclectic musical style that draws on a range of influences and genres spanning electronic   pop   experimental   trip hop   classical and avant garde music ",
]
`;

exports[`clean semantically common use cases 6`] = `
Array [
  "Rapid Evolution  also known as RE   is a software tool for DJs  providing filtering and searching features suitable for musicians ",
  "It can analyze audio files and automatically determine properties such as the musical key  beats per minute  BPM   beat intensity and ReplayGain ↵ It supports file types MP3  MP4  WAV  FLAC  OGG  AAC and APE ",
  "It helps DJs to organize and profile their music  and assists in the process of mixing music by utilizing song metadata to be able to show    ",
]
`;

exports[`clean semantically common use cases 7`] = `
Array [
  "Bjork Gudmundsdottir OTF   bjɜːrk  BYURK  Icelandic   pjoerk ˈkvʏdmʏntsˌtouʰtɪr   About this soundlisten   born 21 November  1965  is an Icelandic singer  songwriter  record producer  actress  and DJ ",
  "Over her four decade career  she has developed an eclectic musical style that draws on a range of influences and genres spanning electronic  pop  experimental  trip hop  classical  and avant garde music ",
]
`;
